crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -90,6 +92,24 @@ impl AccountBuilder {
         self
     }
 
+    /// Adds a designated authentication [`AccountComponent`] to the builder.
+    ///
+    /// This component will be placed at index 0 of the account procedures list.
+    pub fn with_auth_component(mut self, account_component: impl Into<AccountComponent>) -> Self {
+        let component = account_component.into();
+
+        // Validate that the component has exactly one procedure
+        let procedure_count = component
+            .library()
+            .module_infos()
+            .map(|module| module.procedures().count())
+            .sum::<usize>();
+        assert_eq!(procedure_count, 1, "Auth component must have exactly one procedure");
```

Comment:
```suggestion
        assert_eq!(procedure_count, 1, "auth component must have exactly one procedure");
```

Nit: lowercase error messages.